Sources/Build/Triple.swift

```diff
@@ -87,55 +181,193 @@ public struct Triple {
         self.os = os
+        self.osVersion = osVersion
         self.abi = abi ?? .unknown
     }
 
-    fileprivate static func parseOS(_ string: String) -> OS? {
+    fileprivate static func parseOS(_ string: String) throws -> (OS, Version?)? {
+        // Exact match?
+        if let os = OS(rawValue: string) {
+            return (os, nil)
+        }
+
+        // Look for an OS name plus version number
         for candidate in OS.allKnown {
             if string.hasPrefix(candidate.rawValue) {
-                return candidate
+                // Assume the rest of the string is a version number
+                return (candidate, try Version(String(string.dropFirst(candidate.rawValue.count))))
             }
         }
 
         return nil
     }
 
-    public func isDarwin() -> Bool {
-        return vendor == .apple || os == .macOS || os == .darwin
-    }
-
-    public func isLinux() -> Bool {
-        return os == .linux
-    }
-
-    public func isWindows() -> Bool {
-        return os == .windows
-    }
-
-    public static let macOS = try! Triple("x86_64-apple-macosx10.10")
-    public static let x86Linux = try! Triple("x86_64-unknown-linux")
-    public static let ppc64leLinux = try! Triple("powerpc64le-unknown-linux")
-    public static let s390xLinux = try! Triple("s390x-unknown-linux")
-    public static let arm64Linux = try! Triple("aarch64-unknown-linux")
-    public static let armLinux = try! Triple("armv7-unknown-linux-gnueabihf")
-    public static let android = try! Triple("armv7-unknown-linux-androideabi")
-    public static let windows = try! Triple("x86_64-unknown-windows-msvc")
-
-  #if os(macOS)
-    public static let hostTriple: Triple = .macOS
-  #elseif os(Windows)
-    public static let hostTriple: Triple = .windows
-  #elseif os(Linux)
-    #if arch(x86_64)
-      public static let hostTriple: Triple = .x86Linux
-    #elseif arch(powerpc64le)
-      public static let hostTriple: Triple = .ppc64leLinux
-    #elseif arch(s390x)
-      public static let hostTriple: Triple = .s390xLinux
-    #elseif arch(arm64)
-      public static let hostTriple: Triple = .arm64Linux
-    #elseif arch(arm)
-      public static let hostTriple: Triple = .armLinux    
-    #endif
-  #endif
+    // TODO: The "current" triple is not necessarily the desired host triple,
+    // i.e. in cases of 32-bit code running on a 64-bit OS, but this is good
+    // enough for now in practically all relevant cases.
+    public static let hostTriple = Triple.current
+}
+
+extension Triple {
+    /// Creates a triple with the given architecture, OS and OS version.
+    /// The vendor and ABI fields are determined automatically.
+    public static func create(arch: Arch, os: OS, osVersion: Version? = nil) -> Triple {
+        let vendor: Vendor = {
+            switch os {
+            case .Darwin, .iOS, .macOS, .tvOS, .watchOS:
+                return .Apple
+            case .FreeBSD, .Haiku, .Linux, .Windows:
+                return .PC
```

Comment:
This is a behavior change from before, where we used "unknown". Are you sure this is correct?

Reply:
This is fun actually.  It depends on the Linux distribution.  `pc`  is the legacy vendor that is used by default on most i686 targets.  `unknown` is the proper vendor technically.